### PR TITLE
Django social account sso

### PR DESCRIPTION
**Pull request by ddx-day**
Issue: https://github.com/opencv/cvat/issues/1217

Currently there are a few proposals for SSO authentication to bypass the current user/password login on the UI. By using Django social accounts it is also possible to use SSO on the API, retrieving the security token by passing the code from the OAuth2 workflow. This is an example using Amazon Cognito, but any other social account could also be added.

### Motivation and context
Currently CVAT has no functionality to log in with SSO. Other current proposals bypass the current Django framework to add SSO in the UI only, but still use username and password for the API. Using Django social accounts integrates SSO with the API as well, allowing it to be used as an alternative to the username and password, but can also be used together with other SSO frameworks that are UI only. 

### How has this been tested?
Unit tests for SSO manager in cvat-core and integration test with cvat-sdk for /auth/cognito endpoint.

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into [CHANGELOG](https://github.com/cvat-ai/cvat/blob/develop/CHANGELOG.md) file
- [x] I have updated the [documentation](
  https://github.com/cvat-ai/cvat/blob/develop/README.md#documentation) accordingly
- [x] I have added tests to cover my changes
- [x] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

**Comment by nmanovic** [comment]
@ddx-day , thanks for the contribution! We are working on the improving authentication flow right now. You contribution is on time. Please give us some time to review your code. It can take a while because we just investigating how we want to add it. There are many possibilities.

Are you from a company which uses CVAT internally? I would love to meet and collaborate.

**Review by ddx-day — COMMENTED** [review]
Looks like the change to allow email to be used for login happened when this merge happened and that removing the setting actually breaks that change. I am testing it locally and I think I can just revert to the original 'user_email' setting

File: cvat/apps/iam/adapters.py [hunk 1]
```diff
@@ -46,3 +49,21 @@ def get_callback_url(self, request, app):
...
+        return settings.COGNITO_REDIRECT_URI
+
+    def complete_login(self, request, app, access_token, **kwargs):
+        headers = {
+            "Authorization": "Bearer {}".format(access_token),
+        }
+        if 'https' in self.profile_url:
+            extra_data = requests.get(self.profile_url, headers=headers)
+        else:
+            user_url = self.profile_url.lower()
```
Comment (Marishka17) [inline cvat/apps/iam/adapters.py:64]: @ddx-day, Could you please explain why we need to handle this case separately?

**Comment by Marishka17** [comment]
@nmanovic, PR is ready.

**Review by nmanovic — APPROVED** [review]
LGTM

**Pull request merged**